WalletWasabi/BitcoinCore/Mempool/MempoolMirror.cs

```diff
@@ -0,0 +1,142 @@
+using Microsoft.Extensions.Hosting;
+using NBitcoin;
+using NBitcoin.Protocol;
+using NBitcoin.RPC;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.BitcoinCore.Rpc;
+using WalletWasabi.Blockchain.Transactions;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.BitcoinCore.Mempool
+{
+	public class MempoolMirror : PeriodicRunner
+	{
+		/// <param name="period">How often to mirror the mempool.</param>
+		public MempoolMirror(TimeSpan period, IRPCClient rpc, P2pNode node) : base(period)
+		{
+			Rpc = rpc;
+			Node = node;
+		}
+
+		public IRPCClient Rpc { get; }
+		public P2pNode Node { get; }
+		private Dictionary<uint256, Transaction> Mempool { get; } = new();
+		private object MempoolLock { get; } = new();
+
+		public override Task StartAsync(CancellationToken cancellationToken)
+		{
+			Node.MempoolService.TransactionReceived += MempoolService_TransactionReceived;
+			return base.StartAsync(cancellationToken);
+		}
+
+		protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+		{
+			var sw = Stopwatch.StartNew();
+			int added = await MirrorMempoolAsync(stoppingToken).ConfigureAwait(false);
+			Logger.LogInfo($"{added} transactions were copied from the full node to the in-memory mempool within {sw.Elapsed.TotalSeconds} seconds.");
+
+			await base.ExecuteAsync(stoppingToken).ConfigureAwait(false);
+		}
+
```

Comment:
Move ActionAsync method here, to show that is what will be run periodically - increase readability.